python/pyarrow/pandas_compat.py

```diff
@@ -966,7 +966,7 @@ def _extract_index_level(table, result_table, field_name,
 
     if i == -1:
         # The serialized index column was removed by the user
-        return table, None, None
+        return result_table, None, None
```

Comment:
So as long as this method was called once or columns were not missing, this was fine. But if an index column was missing after another index column was already converted to an Index and removed from `result_table`, then `table` and `result_table` is no longer the same.